read-fonts/src/tables/gsub/closure.rs

```diff
@@ -49,13 +53,44 @@ impl<'a> Gsub<'a> {
     }
 
     fn closure_glyphs_once(&self, glyphs: &mut HashSet<GlyphId>) -> Result<(), ReadError> {
+        let lookups_to_use = self.find_reachable_lookups(glyphs)?;
         let lookup_list = self.lookup_list()?;
-        for lookup in lookup_list.lookups().iter() {
+        for (i, lookup) in lookup_list.lookups().iter().enumerate() {
+            if !lookups_to_use.contains(&(i as u16)) {
+                continue;
+            }
             let subtables = lookup?.subtables()?;
             subtables.add_reachable_glyphs(glyphs)?;
         }
         Ok(())
     }
+
+    fn find_reachable_lookups(&self, glyphs: &HashSet<GlyphId>) -> Result<HashSet<u16>, ReadError> {
```

Comment:
Should we also capture lookups that are reachable from feature variations? Since we're not doing instancing (yet) I guess they would all be added unconditionally.

Reply:
good call... I think I'll do this as followup since it stands alone and doesn't need to clutter this up more.